connector/src/main/java/org/geysermc/connector/network/translators/block/entity/BedBlockEntityTranslator.java

```diff
@@ -0,0 +1,41 @@
+package org.geysermc.connector.network.translators.block.entity;
+
+import com.github.steveice10.mc.protocol.data.game.entity.metadata.Position;
+import com.github.steveice10.mc.protocol.data.game.world.block.BlockState;
+import com.nukkitx.math.vector.Vector3i;
+import com.nukkitx.nbt.CompoundTagBuilder;
+import org.geysermc.connector.network.session.GeyserSession;
+import org.geysermc.connector.network.translators.block.BlockTranslator;
+import org.geysermc.connector.utils.BlockEntityUtils;
+
+import java.util.concurrent.TimeUnit;
+
+public class BedBlockEntityTranslator {
```

Comment:
We already have a `BlockEntityTranslator` class that you should be extending here. See the other block entity translator classes.

Reply:
I don't believe this is possible; `BlockEntityTranslator`'s `translateTag` function does not have access to namespace IDs linked to block entities. 
Beds are one of the few exceptions where Bedrock Edition has additional tags over Java Edition. In the future it seems the skull heads will also run into the same problem so it might be wise to modularize this code. Reference: https://minecraft.gamepedia.com/Block_entity

Reply:
That's probably a good call. For the time being, the current method will work, but once we do start moving onto those block entities, I'll change up the code.